common/src/test/java/org/apache/celeborn/common/network/sasl/CelebornSaslSuiteJ.java

```diff
@@ -0,0 +1,209 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.celeborn.common.network.sasl;
+
+import static org.apache.celeborn.common.network.sasl.SaslUtils.*;
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+
+import java.nio.ByteBuffer;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.apache.celeborn.common.CelebornConf;
+import org.apache.celeborn.common.network.TransportContext;
+import org.apache.celeborn.common.network.client.RpcResponseCallback;
+import org.apache.celeborn.common.network.client.TransportClient;
+import org.apache.celeborn.common.network.client.TransportClientBootstrap;
+import org.apache.celeborn.common.network.protocol.RequestMessage;
+import org.apache.celeborn.common.network.server.BaseMessageHandler;
+import org.apache.celeborn.common.network.server.TransportServer;
+import org.apache.celeborn.common.network.util.TransportConf;
+import org.apache.celeborn.common.util.JavaUtils;
+
+/**
+ * Jointly tests {@link CelebornSaslClient} and {@link CelebornSaslServer}, as both are black boxes.
+ */
+public class CelebornSaslSuiteJ {
+  private static final String TEST_USER = "appId";
+  private static final String TEST_SECRET = "secret";
+
+  @BeforeClass
+  public static void setup() {
+    SecretRegistryImpl.getInstance().register(TEST_USER, TEST_SECRET);
+  }
+
+  @Test
+  public void testDigestMatching() {
+    CelebornSaslClient client =
+        new CelebornSaslClient(
+            DIGEST_MD5,
+            DEFAULT_SASL_CLIENT_PROPS,
+            new CelebornSaslClient.ClientCallbackHandler(TEST_USER, TEST_SECRET));
+    CelebornSaslServer server =
+        new CelebornSaslServer(
+            DIGEST_MD5,
+            DEFAULT_SASL_SERVER_PROPS,
+            new CelebornSaslServer.DigestCallbackHandler(
+                userName -> assertEquals(TEST_USER, userName), SecretRegistryImpl.getInstance()));
+
+    assertFalse(client.isComplete());
+    assertFalse(server.isComplete());
+
+    byte[] clientMessage = client.firstToken();
+
+    while (!client.isComplete()) {
+      clientMessage = client.response(server.response(clientMessage));
+    }
+    assertTrue(server.isComplete());
+
+    // Disposal should invalidate
+    server.dispose();
+    assertFalse(server.isComplete());
+    client.dispose();
+    assertFalse(client.isComplete());
+  }
+
+  @Test
+  public void testDigestNonMatching() {
+    CelebornSaslClient client =
+        new CelebornSaslClient(
+            DIGEST_MD5,
+            DEFAULT_SASL_CLIENT_PROPS,
+            new CelebornSaslClient.ClientCallbackHandler(TEST_USER, "my-secret"));
```

Comment:
nit: to make it explicit - mutate `TEST_SECRET` so that it does not match. For example: `"invalid-" + TEST_SECRET` or some such ?